src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_traces_overview/sub_components/http_status_code/use_get_http_status_color.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useEuiTheme } from '@elastic/eui';
+
+export const useGetHttpStatusColor = (status: string | number) => {
+  const { euiTheme } = useEuiTheme();
+  const httpStatusCodeColors: Record<string, string> = {
+    1: euiTheme.colors.vis.euiColorVisGrey0,
+    2: euiTheme.colors.vis.euiColorVisSuccess0,
+    3: euiTheme.colors.vis.euiColorVisGrey0,
+    4: euiTheme.colors.vis.euiColorVisWarning0,
+    5: euiTheme.colors.vis.euiColorVisDanger0,
+  };
+  const intStatus = typeof status === 'string' ? parseInt(status.replace(/\D/g, ''), 10) : status;
+  return httpStatusCodeColors[intStatus.toString().substring(0, 1)];
```

Comment:
Maybe?
```suggestion
  const firstDigit = String(status).charAt(0);
  return httpStatusCodeColors[firstDigit];
```

Also, we might want to return something in case the status code is invalid. (7xx is not covered here either)

Reply:
https://github.com/elastic/kibana/pull/210612/commits/b7911b04181d8ff8b10d62f98addb16f56887ae1